### PR TITLE
Fixed grammatical error in "enum specifier" error msg #94443

### PR DESCRIPTION
**Pull request by kper**
As discussed in #94443, this PR changes the wording to be more correct.

### CODE REVIEW

File: clang/include/clang/Basic/DiagnosticSemaKinds.td [hunk 1]
```diff
@@ -6088,9 +6088,9 @@ def err_redefinition_different_concept : Error<
   "redefinition of concept %0 with different template parameters or requirements">;
 def err_tag_reference_non_tag : Error<
   "%select{non-struct type|non-class type|non-union type|non-enum "
-  "type|typedef|type alias|template|type alias template|template "
-  "template argument}1 %0 cannot be referenced with a "
-  "%select{struct|interface|union|class|enum}2 specifier">;
+  "type|typedef|type alias|template|alias template|template "
```
Comment (Endilll) [inline clang/include/clang/Basic/DiagnosticSemaKinds.td:6091]: Why are you removing "type" from "type alias template"? Why are you keeping "type" in "type alias"?

Comment (sdkrystian) [reply clang/include/clang/Basic/DiagnosticSemaKinds.td:6091]: I think this is using the standard term "alias template" as defined in [[temp.alias] p1](http://eel.is/c++draft/temp.alias#1). As for "type alias", the standard doesn't refer to _alias-declarations_ by any other name. I suppose the pedantically correct phrasing would be to refer to `typedef`s and _alias-declarations_ as _typedef-names_.

Comment (Endilll) [reply clang/include/clang/Basic/DiagnosticSemaKinds.td:6091]: The change makes sense now, thank you.

**Review by Endilll — APPROVED** [review]

**Review by AaronBallman — APPROVED** [review]
LGTM

**Comment by Rajveer100** [comment]
I think this can be merged considering the approved changes?

**Pull request merged**